analyzers/PapasSim.py

```diff
@@ -67,8 +68,8 @@ def process(self, event):
         pfsim_particles = self.simulator.ptcs
 
         #these are the particles before simulation
-        simparticles = sorted(pfsim_particles,
-                              key=lambda ptc: ptc.e(), reverse=True)     
+        simparticles = sorted(pfsim_particles, key=P4.sort_key)
```

Comment:
why not reverse=True? are you sure the particles are sorted by decreasing energy or pT?